### DISCUSSION TITLE
Apply cors rules with "ExposeHeader"

### DISCUSSION THREAD

**Discussion by cpoyatos1** [digitalocean/terraform-provider-digitalocean#1015] [Q&A] [answered]
Hello community,

I have a HCL definition for a space bucket:
```hcl
resource "digitalocean_spaces_bucket" "bucket" {
  name   = "bucket-${random_uuid.bucket-identifier.result}"
  region = "fra1"

  lifecycle_rule {
    id                                     = random_uuid.bucket-lifecycle_rule-identifier.result
    enabled                                = true
    abort_incomplete_multipart_upload_days = 1
  }

  cors_rule {
    allowed_headers = ["*"]
    allowed_methods = ["GET", "PUT", "POST", "DELETE", "HEADER"]
    allowed_origins = ["*"]
    max_age_seconds = 3000
  }
}
```

**About my problem:**
I have created a GraphQL backend that generates pre-signed multipart upload URLs. For a multipart upload I need the etag from the response header. I am using axios as the http client. But because of CORS I can't read the etag. I have to configure the "ExposeHeader" in the CORS settings.

I have tried this successfully using these documentation: https://docs.digitalocean.com/products/spaces/how-to/configure-cors/#xml

I would like to be able to configure this via Terraform as well. Do you have any idea how to do this? According to the [digitalocean provider documentation](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/resources/spaces_bucket#cors_rule), this is not supported.

**Comment by danaelhe** [comment]
Thanks for the write up! We are investigating internally and will respond as soon as we can.

**Reply by cpoyatos1** [reply]
Thanks!

**Comment by danaelhe** [comment]
We believe we can support this by creating a `digitalocean_spaces_bucket_cors_configuration` resource. Similar to the aws_provider support [here](https://registry.terraform.io/providers/-/aws/latest/docs/resources/s3_bucket_cors_configuration#example-usage).

I've created a ticket in our backlog and we'll prioritize it and implement it as soon as we can. If you need this support immediately, we also always welcome and deeply appreciate PRs if you'd like to take a crack at it yourself :)

**Reply by cpoyatos1** [reply]
Thanks a lot!

**Comment by danaelhe** [comment, marked as answer]
Addressed in this [PR](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1021)!
